dotcom-rendering/src/lib/getFrontsAdPositions.ts

```diff
@@ -9,10 +13,66 @@ type GroupedCounts = {
 	standard: number;
 };
 
+type AdCandidate = Pick<DCRCollectionType, 'collectionType'>;
+
+const getMerchHighPosition = (collectionCount: number): number =>
+	collectionCount >= 4 ? 2 : 0;
+
+// This happens on the recipes front, where the first container is a thrasher see: https://github.com/guardian/frontend/pull/20617
+const hasFirstContainerThrasher = (collectionType: string, index: number) =>
+	index === 0 && collectionType === 'fixed/thrasher';
+
+const hasAdjacentCommercialContainer = (
+	collectionIndex: number,
+	merchHighPosition: number,
+) => collectionIndex === merchHighPosition;
+
+const hasAdjacentThrasher = (index: number, collections: AdCandidate[]) =>
+	collections[index + 1]?.collectionType === 'fixed/thrasher';
+
+const isMostViewedContainer = (collection: AdCandidate) =>
+	collection.collectionType === 'news/most-popular';
+
+const shouldInsertAd =
+	(merchHighPosition: number) =>
+	(
+		collection: AdCandidate,
+		collectionIndex: number,
+		collections: AdCandidate[],
+	) =>
+		!(
+			hasFirstContainerThrasher(
+				collection.collectionType,
+				collectionIndex,
+			) ||
+			hasAdjacentCommercialContainer(
+				collectionIndex,
+				merchHighPosition,
+			) ||
+			hasAdjacentThrasher(collectionIndex, collections) ||
+			isMostViewedContainer(collection)
+		);
+
+const isEvenIndex = (_collection: unknown, index: number) => index % 2 === 0;
+
 /**
- * The maximum number of fronts-banner ads that can be inserted on any front.
+ * We do not insert mobile ads:
+ * a. after the first container if it is a thrasher
+ * b. after a commercial container (e.g. merchandising-high)
+ * c. between a regular container and a thrasher
+ * d. after the Most Viewed container.
+ * After we've filtered out the containers next to which we can insert an ad,
+ * we take every other container, up to a maximum of 10, for targeting MPU insertion.
  */
-const MAX_FRONTS_BANNER_ADS = 6;
+const getMobileAdPositions = (collections: AdCandidate[]): number[] => {
+	const merchHighPosition = getMerchHighPosition(collections.length);
+
+	return collections
+		.filter(shouldInsertAd(merchHighPosition))
+		.filter(isEvenIndex)
+		.map((collection: AdCandidate) => collections.indexOf(collection))
```

Comment:
I believe `indexOf` returns a -1 when the element isn't found, do we need to filter those out here or is it okay to not? Asking as you mentioned adding the filtering to `getTagFrontMobileAdPositions`.

Reply:
https://github.com/guardian/dotcom-rendering/pull/9415#discussion_r1383713411